Sources/MongoSwift/MongoDatabase.swift

```diff
@@ -272,6 +272,103 @@ public class MongoDatabase {
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      * - Note: Supported in MongoDB version 4.0+ only.
+      */
+     public func watch<T: Codable>(_ pipeline: [Document] = [],
+                                   options: ChangeStreamOptions? = nil,
+                                   session: ClientSession? = nil,
+                                   withFullDocumentType: T.Type) throws ->
+                                   ChangeStream<ChangeStreamDocument<T>> {
+        return try self.watch(pipeline,
+                              options: options,
+                              session: session,
+                              withReturnType: ChangeStreamDocument<T>.self)
+     }
+
+     /**
+      * Starts a `ChangeStream` on a database. Excludes system collections. Associates the specified `Codable` type `T`
+      * with the returned `ChangeStream`.
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use when constructing the `ChangeStream`.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      *   - withReturnType: The type that the entire change stream response will be decoded to.
+      * - Returns: A `ChangeStream` on all collections in a database.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+      *   - `ServerError.commandError` if the pipeline passed is invalid.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      *   - `DecodingError` if an error occurs while decoding user-defined `withReturnType` `Codable` type.
+      * - SeeAlso:
+      *   - https://docs.mongodb.com/manual/changeStreams/
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      * - Note: Supported in MongoDB version 4.0+ only.
+      */
+     public func watch<T: Codable>(_ pipeline: [Document] = [],
+                                   options: ChangeStreamOptions? = nil,
+                                   session: ClientSession? = nil,
+                                   withReturnType: T.Type) throws ->
+                                   ChangeStream<T> {
+        let pipeline: Document = ["pipeline": pipeline]
+        let connection = try self._client.connectionPool.checkOut()
+        let opts = try encodeOptions(options: options, session: session)
+        return try self.withMongocDatabase(from: connection) { dbPtr in
+            let changeStreamPtr: OpaquePointer = mongoc_database_watch(dbPtr, pipeline._bson, opts?._bson)
+            return try ChangeStream<T>(stealing: changeStreamPtr,
```

Comment:
According to the [Updating the Cached Resume Token](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#updating-the-cached-resume-token) section of the spec, in the cases of `startAfter` or `resumeAfter`, we need to set the cached token to one we use to resume/start after. I think that logic can be put into the initializer of `ChangeStream` so it'll get transitively included here and all other `watch` functions.

On a somewhat related note, the `resumeToken` field of a `ChangeStream` should be optional, since it won't always be set on creation. 

And one final somewhat related note: `init(resumeToken:)` on `ResumeToken` can have the label omitted since it is redundant (i.e. `init(_ resumeToken: Document)`).